hypothesis/viahtml
# Integrate Checkmate as the blocking provider in ViaHTML

For: https://github.com/hypothesis/checkmate/issues/3

This strips out the existing blocklist code and replaces it with a client for Checkmate. ~If this all works, I think we might want to think about moving this out to a library at some point.~

~This whole business is intended to be totally separate from `viahtml` so we can pluck it off as a separate library when it's a bit more battle tested. The bigger this gets the less we want to copy paste it everywhere.~ Done

Some things it would be nice to add to this:

 * Helpers for handling any redirection we might want to issue from Checkmate
 * Maybe some helpers for filtering which kinds or responses you care about?
 * Maybe some caching on this side? No need to request the same URL over and over if we already know

~Currently ViaHTML is the only Python 3 client, but I can see us wanting to talk to this in other products. We already want it in Via, but I don't know how hard it is to write something which is Python 2 / 3 compatible. I've not tried that before.~ Done

### Review notes

 * ~Should we just bit the bullet and put this in a separate client library now? It's not that much work and we have that fairly automated now.~ Done
 * ~Should we investigate having Checkmate be an app and provide a client for itself? I suspect it's possible and has some nice features~ Not done, but I still like the idea

### Testing notes

 * Run `make dev`
 * Goto: http://localhost:9085/proxy/https://bad.example.com/
 * You should some error as the page isn't real
 * In the logs you should see `[WARNING]: Failed to check URL against Checkmate: Cannot connect to service`
 * Start checkmate service with `make dev`
 * Goto: http://localhost:9085/proxy/https://bad.example.com/
 * You should now see the malicious blocking page

### Release notes

**Do not merge this until monitoring etc. is in place for Checkmate: https://github.com/hypothesis/checkmate/issues/7**
